components/experimental/src/units/converter_factory.rs

```diff
@@ -168,10 +183,22 @@ impl ConverterFactory {
         ///     For example, `newton` has the basic units:  `gram`, `meter`, and `second` (each one has it is own power and si prefix).
         fn insert_non_basic_units(
             factory: &ConverterFactory,
-            units: &[SingleUnit],
+            units: &SingleUnitVec,
             sign: i16,
             map: &mut LiteMap<u16, PowersInfo>,
-        ) -> Option<()> {
+        ) -> Result<(), InvalidUnitError> {
+            let units: &[&SingleUnit] = match units {
+                SingleUnitVec::Zero => &[],
+                SingleUnitVec::One(unit) => &[unit],
+                SingleUnitVec::Two(unit1, unit2) => &[unit1, unit2],
+                #[cfg(feature = "alloc")]
+                SingleUnitVec::Multi(units) => &units.iter().collect::<Vec<_>>(),
+                #[cfg(not(feature = "alloc"))]
+                _ => return Err(InvalidUnitError),
+            };
```

Comment:
Suggestion: This should be a helper fn on SingleUnitVec. `pub fn as_slice(&self) -> &[SingleUnit]`

To make that fn work, you will need to change the Two variant to store `([SingleUnit; 2])` instead of `(SingleUnit, SingleUnit)`.

Reply:
Here, I could not implement `as_slice`, because in the case of `One` and `Two`, I could not return `&[SingleUnit]` because there is no locale container holds the value

do you have suggestions ?

Reply:
thanks for the info, done :)